src/node.rs

```diff
@@ -516,43 +535,51 @@ impl PbftNode {
                     Ok(msgs)
                 })?;
 
-        // Sync local state with actual state. We re-sync the sequence number with head
-        // again, even though we already did that in `on_block_new`, since if we're
-        // catching up as the primary, the sequence number should actually just be
-        // the head's block num, and not that+1.
-        if state.seq_num != head.block_num {
-            info!(
-                "Updating sequence number from {} to {}.",
-                state.seq_num, head.block_num
-            );
-            state.seq_num = head.block_num;
-        }
-
         let view = messages[0].info().get_view();
         if state.view != view {
             info!("Updating view from {} to {}.", state.view, view);
             state.view = view;
         }
 
+        // Add messages to backlog so that `handlers::commit` can process a commit
+        // message normally
         for message in &messages {
             self.msg_log.add_message(message.clone());
         }
 
-        // Commit the new block, using one of the parsed commit messages to simulate
+        // Commit the new block, using one of the parsed messages to simulate
         // having received a regular commit message.
-        handlers::commit(state, &mut self.msg_log, &mut *self.service, &messages[0])?;
+        handlers::commit(
+            state,
+            &mut self.msg_log,
+            &mut *self.service,
+            &messages[0].as_msg_type(PbftMessageType::Commit),
+        )?;
 
         // Start a view change if we need to force one for fairness
         if state.at_forced_view_change() {
             self.force_view_change(state);
         }
 
+        // Ensure that the message that we generated from the BlockNew message
+        // has the right sequence number.
+        let mut fixed_msg = msg.clone();
```

Comment:
Self-contained change

Reply:
There isn't really a single bug that each of these changes fix, so I don't see what splitting it up gives us - the commit message would just be something like "part of making catchup more robust".

Reply:
That's okay, 1 commit != 1 bug fix. Generally speaking, 1 PR should be 1 bug fix. Best practice from my experience is that the smallest self-contained change that results in runnable, improved code should be its own commit. Each of these changes is distinct; they are all in service of fixing the same bug, but they are distinct changes nonetheless. For this change, the comment you provided is already a pretty good commit message. In my opinion, it's a lot easier to understand what's going on when each of the changes is described, rather than all being lumped together and just saying "this makes it more robust."